src/new-client/src/utils/__tests__/emitTooLong.test.ts

```diff
@@ -0,0 +1,35 @@
+import { emitTooLongMessage } from "utils/emitTooLong"
+import { testScheduler } from "utils/testScheduler"
+
+const TOO_LONG = { status: "too long" }
+
+const TAKING_TOO_LONG = 2_000
+
+const response = {
+  response: "I am a response",
+}
+
+describe("utils/emitTooLong", () => {
+  it("should emit normal response if response is faster than timer", () => {
+    testScheduler().run(({ expectObservable, cold }) => {
+      const input = cold("    ---(a|)", { a: response })
+      const expectedOutput = "---(b|)"
```

Comment:
```suggestion
      const input = cold("    2s (a|)", { a: response })
      const expectedOutput = "2s (b|)"
```

It's better if we test the limit :slightly_smiling_face: